swc_ecma_parser : Cannot compile with --no-default-features

### Describe the bug

cargo add swc_ecma_parser --no-default-features

cargo build

```
error[E0599]: no associated function or constant named `is_flow_reserved_type_name` found for struct `parser::Parser<I>` in the current scope 
    --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\class_and_fn.rs:1899:34
     |
1899 |                         if Self::is_flow_reserved_type_name(&ident.sym) {
     |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^ associated function or constant not found in `parser::Parser<I>`
     |
    ::: C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\mod.rs:66:1
     |
  66 | pub struct Parser<I: self::input::Tokens> {
     | ----------------------------------------- associated function or constant `is_flow_reserved_type_name` not found for this struct
     |
note: if you're trying to build a new `parser::Parser<I>`, consider using `parser::Parser::<I>::new_from` which returns `parser::Parser<_>`
    --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\mod.rs:181:5
     |
 181 |     pub fn new_from(mut input: I) -> Self {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `emit_flow_reserved_type_name_error` found for mutable reference `&mut parser::Parser<I>` in the current scope  
   --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\module_item.rs:206:38
    |
206 | ...                   self.emit_flow_reserved_type_name_error(
    |                       -----^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut parser::Parser<I>`

error[E0599]: no method named `emit_flow_reserved_type_name_error` found for mutable reference `&mut parser::Parser<I>` in the current scope  
   --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\module_item.rs:211:34
    |
211 | ...                   self.emit_flow_reserved_type_name_error(local.span, &local.sym);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut parser::Parser<I>`

error[E0599]: no method named `emit_flow_reserved_type_name_error` found for mutable reference `&mut parser::Parser<I>` in the current scope  
   --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\module_item.rs:329:30
    |
329 |                         self.emit_flow_reserved_type_name_error(local.span, &local.sym);
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut parser::Parser<I>`

error[E0599]: no method named `emit_flow_reserved_type_name_error` found for mutable reference `&mut parser::Parser<I>` in the current scope  
   --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\module_item.rs:354:26
    |
354 |                     self.emit_flow_reserved_type_name_error(local.span, &local.sym);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut parser::Parser<I>`

error[E0599]: no method named `emit_flow_reserved_type_name_error` found for mutable reference `&mut parser::Parser<I>` in the current scope  
   --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\module_item.rs:985:26
    |
985 |                     self.emit_flow_reserved_type_name_error(local.span, &local.sym);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut parser::Parser<I>`

error[E0599]: no method named `emit_flow_reserved_type_name_error` found for mutable reference `&mut parser::Parser<I>` in the current scope  
    --> C:\Users\Admin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\swc_ecma_parser-41.1.0\src\parser\module_item.rs:1042:26
     |
1042 |                     self.emit_flow_reserved_type_name_error(local.span, &local.sym);
     |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut parser::Parser<I>`

For more information about this error, try `rustc --explain E0599`.                                                                           
error: could not compile `swc_ecma_parser` (lib) due to 7 previous errors                              
```

### Input code

```typescript

```

### Config

```json

```

### Link to the code that reproduces this issue

https://play.swc.rs/?version=1.15.41&code=H4sIAAAAAAAAA0vOzyvOz0nVy8lP11AqLk9W0gQADuvaOBIAAAA%3D&config=H4sIAAAAAAAAA1WPSw7DIAwF9zkF8rrbdtE79BAWdSIifrKJVBTl7iUE0maH3xsz8jooBbNoeKq1PMsQkYX4nEsi2Sf8lARIOxTNJia49XaWvRrRCtVoOxpIyBOluiX3hoMNQajjLXPGmzH%2FC3VwkUnkCu4o%2BsnSVTc0JbjwXmrZDkk50qF%2FwA%2FqsvNjMPLqm4kXGrYvhlQioBQBAAA%3D

### SWC Info output

_No response_

### Expected behavior

No error

### Actual behavior

_No response_

### Version

swc_ecma_parser v41.1.0

### Additional context

_No response_

--- a/.changeset/fix-parser-no-default-flow.md
+++ b/.changeset/fix-parser-no-default-flow.md
@@ -1,0 +1,6 @@
+---
+swc_core: patch
+swc_ecma_parser: patch
+---
+
+fix(es/parser): allow compilation without default features

--- a/crates/swc_ecma_parser/src/parser/typescript_stubs.rs
+++ b/crates/swc_ecma_parser/src/parser/typescript_stubs.rs
@@ -1,16 +1,23 @@
 //! Stub implementations for TypeScript parsing methods when the `typescript`
 //! feature is disabled.
 //!
-//! These methods are never called at runtime (the calls are guarded by
-//! `syntax().typescript()` which returns `false` when the feature is disabled),
-//! but Rust needs them to exist for compilation to succeed.
+//! Shared parser modules call these methods even when the `typescript` feature
+//! is disabled. Runtime syntax checks make TypeScript and Flow behavior
+//! unavailable in this configuration, but Rust still needs matching method
+//! definitions for compilation to succeed.
 
-use swc_common::BytePos;
+use swc_common::{BytePos, Span};
 use swc_ecma_ast::*;
 
 use crate::{input::Tokens, lexer::Token, PResult, Parser};
 
 impl<I: Tokens> Parser<I> {
+    pub(super) fn is_flow_reserved_type_name(_name: &str) -> bool {
+        false
+    }
+
+    pub(super) fn emit_flow_reserved_type_name_error(&mut self, _span: Span, _name: &str) {}
+
     pub(crate) fn try_parse_ts<T, F>(&mut self, _op: F) -> Option<T>
     where
         F: FnOnce(&mut Self) -> PResult<Option<T>>,

--- a/tests.yml
+++ b/tests.yml
@@ -40,6 +40,8 @@ check:
         - "cargo hack check --feature-powerset --exclude-features __rkyv"
     swc_ecma_loader:
         - "cargo hack check --feature-powerset"
+    swc_ecma_parser:
+        - "cargo check --no-default-features"
     swc_ecma_transforms:
         # - "cargo hack check --feature-powerset"
         - "cargo check --features concurrent --features par-core/chili"

